src/Foreign/Marshal/Alloc/Compat.hs

```diff
@@ -0,0 +1,44 @@
+#if !MIN_VERSION_base(4,8,0)
+{-# LANGUAGE ForeignFunctionInterface #-}
+#endif
+module Foreign.Marshal.Alloc.Compat (
```

Comment:
Now that we have `Foreign.Marshal` re-exports, both `Foreign.Marshal.Alloc` and `Foreign.Marshal.Array`.  So now that we have `Compat` version of those modules, I think it would be nice to have `Foreign.Marshal.Compat` too, that re-exports the `Compat` versions.  But we could do that as a separate change.